### DISCUSSION TITLE
Difference between schema creation by using Object category definition API and schema.json file

### DISCUSSION THREAD

**Discussion by Dhanashree-Patil95** [Sunbird-Knowlg/Community#157] [Q&A]
we want to create Domain specific schema for ex. Job schema having fields like job_title, salary ,joining_date. There seem to be 2 ways to define a new schema in knowledge platform, using object category definition API and schema.json file similar in 5.1.0 version of knowledge platform there is Schema folder containing schema.json and config.json files for various objects like question , questionset , So can anyone help to explain how these two methods are different ? @Krishnaj20

**Comment by pallakartheekreddy** [comment]
@Dhanashree-Patil95 In Knowlg, you have two methods for extending existing object schemas, such as content, collection, question, question set, and more:

**Object Category Definition API:** This method involves using the Object Category Definition API to extend the schema. It utilizes the existing base schema, which is represented by the `schema.json` JSON file for a particular object, to make the schema extensions.

**Updating `schema.json`:** The second approach is to update the `schema.json` file directly with new properties to extend the schema. This allows you to define and add new properties to the base schema.

Both of these methods provide flexibility for customizing and expanding the schema definitions as needed for different objects in Knowlg.

**Reply by pallakartheekreddy** [reply]
@Dhanashree-Patil95 Hope your query is now resolved. Please mark the thread as answered

**Reply by pallakartheekreddy** [reply]
@Dhanashree-Patil95 As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
